pkg/ccl/partitionccl/partition.go

```diff
@@ -115,7 +115,7 @@ func valueEncodePartitionTuple(
 			return nil, err
 		}
 		value, err = rowenc.EncodeTableValue(
-			value, descpb.ColumnID(encoding.NoColumnID), datum, scratch,
+			value, descpb.NoColumnID, datum, scratch,
```

Comment:
I've been thinking that these ids are getting annoying from a dependency hierarchy perspective. I hate the relationship to `tree` too. I don't know exactly where to slot it, but I want the ID types to be extremely low in the import tree and then, maybe, get forwarded up to `catalog` or something. What I have in mind is a `catid` package which can be imported by tree, descpb, etc. I'm running into this now because I want to move the new schema changer state (`scpb`) below `descpb` so it can be imported and am being thwarted by these IDs. I'll make that change after this goes in and would love to move `encoding.NoColumnID` to `catid.NoColumnID` or perhaps even `catid.NoColumn`. Thoughts?